requests/models.py

```diff
@@ -914,14 +914,17 @@ def raise_for_status(self):
 
         http_error_msg = ''
         if isinstance(self.reason, bytes):
-            # We attempt to decode utf-8 first because some servers
-            # choose to localize their reason strings. If the string
-            # isn't utf-8, we fall back to iso-8859-1 for all other
-            # encodings. (See PR #3538)
-            try:
-                reason = self.reason.decode('utf-8')
-            except UnicodeDecodeError:
-                reason = self.reason.decode('iso-8859-1')
+            if is_py2:
+                reason = self.reason.decode('utf-8', 'ignore')
```

Comment:
If `IOError` will not accept unicode on Python 2, why are we bothering to decode it at all here?

Reply:
Don't really know, only try to get back to pre- #3554 situation. I'm not good about encoding & such...

You can reject this PR as requests is OK with Python 3, and it's 2018.

Reply:
Sorry, let me try to be clearer:

```py
>>> # Python 2
>>>'foo'.decode('utf-8')
u'foo'
```

```py
>>> # Python 2
>>> u'foo'.encode('utf-8')
'foo'
```

I think you want `encode` in this case

Reply:
it's like a formula in py2... just, `str.decode -> unicode` & `unicode.encode -> str`